### PR TITLE
WDM-KS teething fix

### PR DESCRIPTION
**Pull request by daschuer**
This branch fixed the waveform issues with WDM-KS.

### CODE REVIEW

**Comment by daschuer** [comment]
Fixes https://bugs.launchpad.net/mixxx/2.1/+bug/1729345

File: src/soundio/sounddeviceportaudio.cpp [hunk 1]
```diff
@@ -347,6 +347,8 @@ SoundDeviceError SoundDevicePortAudio::open(bool isClkRefDevice, int syncBuffers
         qDebug() << "PortAudio: Started stream successfully";
     }
 
+    m_clkRefTimer.start();
```
Comment (rryan) [inline src/soundio/sounddeviceportaudio.cpp]: Should it go inside `if (isClkRefDevice) {` on line 359?

File: src/soundio/sounddeviceportaudio.cpp [hunk 2]
```diff
@@ -999,8 +1001,9 @@ void SoundDevicePortAudio::updateCallbackEntryToDacTime(
     double diff = (timeSinceLastCbSecs + callbackEntrytoDacSecs) -
             (m_lastCallbackEntrytoDacSecs + bufferSizeSec);
 
-    if (timeSinceLastCbSecs < bufferSizeSec * 2 &&
-            fabs(diff) / bufferSizeSec > 0.1) {
+    if (callbackEntrytoDacSecs <= 0 ||
+            (timeSinceLastCbSecs < bufferSizeSec * 2 &&
+            fabs(diff) / bufferSizeSec > 0.1)) {
         // Fall back to CPU timing:
         // If timeSinceLastCbSecs from a CPU timer is reasonable (no underflow)
```
Comment (rryan) [inline src/soundio/sounddeviceportaudio.cpp]: could you update the comment to describe the new case that triggers this logic?

**Comment by rryan** [comment]
lgtm, thanks!

**Comment by daschuer** [comment]
@JosepMaJAZ has confirmed this fixed the issue for him.

**Pull request merged**